fmlcore/src/main/java/net/minecraftforge/fml/config/IConfigEvent.java

```diff
@@ -11,7 +11,7 @@
 import java.util.function.Function;
 
 public interface IConfigEvent {
-    record ConfigConfig(Function<ModConfig, IConfigEvent> loading, Function<ModConfig, IConfigEvent> reloading) {}
+    record ConfigConfig(Function<ModConfig, IConfigEvent> loading, Function<ModConfig, IConfigEvent> reloading, Function<ModConfig, IConfigEvent> unloading) {}
```

Comment:
Do we consider this part of the public API? Because if it is, we may need some sort of bouncer for the old two-arg constructor to avoid a breaking change here.

Reply:
I debated about this. It's definitely not part of the "standard" public API - it's there to allow FMLonly and Forge to share the code. It's arguable that someone in sponge is using it to supply configs in a completely different way, but I don't have any insight into that. IMO we should not treat it as _public_.

Reply:
How are we to handle changes of a breaking nature in this section of the API? Does this section in particular (or the FML API, however nebulous that may be at the moment, as a whole) share none of the breaking changes guarantees that the main Forge API has?

We could certainly push this now and bank on the (high) chance that nobody else other than Forge uses this. But we should be considering how we will treat breaking changes in this part of the API for the future.

Reply:
We already discussed related things before. Now that we are using @ApiStatus.Internal anything without it is covered under the binary breaking rules. So yes this needs a bouncer for the constructor.